### ISSUE TITLE
Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.6:yarn

### ISSUE THREAD

**Issue by achaychenko** [eirslett/frontend-maven-plugin#740]
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.6:yarn (gulp process-resources) on project jira-gadgets-plugin: Failed to run task: 'yarn run process-resources --module jira-gadgets-plugin --skip-minification=false --production=false' failed. java.io.IOException: Cannot run program "/home/alex/jira/jira-gadgets-plugin/target/node/yarn/dist/bin/yarn" (in directory "/home/alex/jira/jira-gadgets-plugin"): error=2, No such file or directory -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.6:yarn (gulp process-resources) on project jira-gadgets-plugin: Failed to run task

OS: Ubuntu 18.04
Maven-3.2.5
java version "9.0.4"

**Comment by ShiftyMcCool** [comment]
Bumping this issue.  Using install-node-and-yarn installs yarn in .../node/yarn/yarn1.13.0/bin/yarn but the plugin is looking in .../node/yarn/dist/bin/yarn

**Comment by eirslett** [comment]
Does it work with an earlier version if yarn? Maybe they changed the installation directory?

**Comment by ShiftyMcCool** [comment]
Sorry for the late reply.  They must have changed the directory name but I haven't tried downgrading the version of Yarn since we're coding a new project in Angular and rely on the latest version to avoid any issues between the latest Angular and old Yarn.  Not sure if you can use a wildcard of some kind to get the install directory, or maybe get the version installed from Yarn itself, but I'm guessing that this plugin will need to support the new directory format to continue working.

**Comment by eirslett** [comment]
Yup, I guess it's easier to just drop support for older versions of yarn, and change the directory layout.

**Comment by aleksandr-wemakesoftware** [comment]
confirming the bug: 
earlier versions (tested with 0.25.2 yarn version) install yarn to 
project_root/node/yarn
yarn versions 1.x are installed under 
project_root/node/yarn/yarn-1.x

Do you need help in resolving this issue?

**Comment by eirslett** [comment]
sure, just make a PR.

**Comment by venkatapatti** [comment]
is it fixed yet? I am still facing the issue with yarn version 1.17.3

**Comment by githoniel** [comment]
I think #845 will help

**Comment by pingsutw** [comment]
Faced the same issue, how do I resolve it

**Comment by ShiftyMcCool** [comment]
@pingsutw I believe if you rename the yarn directory after the first install, it will work.  For example, Yarn installs here .../node/yarn/yarn1.13.0/bin/yarn.  After you get the error the first time, rename "yarn1.13.0" (or whatever your version is) to .../node/yarn/yarn/bin/yarn.  Run the build again and it should detect that yarn is already installed and keep going.  Other that that, you can downgrade to a much older version of yarn

**Comment by pingsutw** [comment]
```
sudo dpkg --configure -a
sudo apt-get -f install
sudo apt-get full-upgrade
sudo apt-get autoremove
sudo apt-get clean
```
After tried these commands, it works for me.
